HybridCar.cpp

```diff
@@ -1,19 +1,29 @@
 #include "HybridCar.hpp"
+#include"Exceptions.hpp"
 #include <iostream>
 
 HybridCar::HybridCar(PetrolEngine* petrolEng, ElectricEngine* electricEng)
-    : petrolEngine_(petrolEng)
-    , electricEngine_(electricEng)
+    : PetrolCar(petrolEng)
+    , ElectricCar(electricEng)
 {
     std::cout << __FUNCTION__ << std::endl;
 }
 
-HybridCar::~HybridCar()       { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::turnLeft()      { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::turnRight()     { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::brake()         { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::accelerate(int) { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::charge()        { std::cout << __FUNCTION__ << std::endl; }
-void HybridCar::refuel()        { std::cout << __FUNCTION__ << std::endl; }
- 
+HybridCar::~HybridCar()
+{ 
+    std::cout << __FUNCTION__ << std::endl; 
+}
+
+PetrolEngine * HybridCar::changeEngine(int power,float capacity,int gears)
+{
+    PetrolCar::changeEngine(power,capacity,gears);
+}
 
+ElectricEngine * HybridCar::changeEngine(int power, int battery)
```

Comment:
można też tych funkcji w changeEngine w ogóle tutaj nie pisać. Ponieważ różnią się one parametrami to odpowiednie wersje od razu zostaną wywołane :)